### PR TITLE
Kernel/PCI: Expose PCI option ROM data from the sysfs interface

### PR DESCRIPTION
**Pull request by supercomputer7**
For each exposed PCI device in sysfs, there's a new node called "rom" and
by reading it, it exposes the raw data of PCI option ROM blob to a user
for examining the blob.

### CODE REVIEW

**Comment by supercomputer7** [comment]
Rebased and force-pushed to re-test CI now it's working again :)

**Comment by supercomputer7** [comment]
I thought about this and came to a conclusion that I should put more effort into this and make a mechanism to ensure that if both userspace and a kernel driver try to load data from the ROM, they will have to take a spinlock of some sort somewhere.

**Comment by supercomputer7** [comment]
Closing this for now until I have more time to work on this :)

**Comment by supercomputer7** [comment]
Let's try this again :)

File: Kernel/FileSystem/SysFS/Subsystems/Bus/PCI/DeviceExpansionROM.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    // there's no meaningful way to handle negative values, so just return with an error.
+    if (offset < 0)
+        return Error::from_errno(EINVAL);
+    auto unsigned_offset = static_cast<size_t>(offset);
+    // NOTE: If the offset is beyond the PCI option ROM size, return EOF.
+    if (unsigned_offset >= m_option_rom_size)
+        return 0;
+
+    auto blob = TRY(try_to_generate_buffer(unsigned_offset, count));
+    if ((size_t)offset >= blob->size())
```
Comment (GMTA) [inline Kernel/FileSystem/SysFS/Subsystems/Bus/PCI/DeviceExpansionROM.cpp:43]: No C-style casts please

**Comment by GMTA** [comment]
Commit message typo: "exteremly"

**Comment by GMTA** [comment]
Apart from those two tiny comments, everything still seems to work with these changes. Have not tested actually dumping the ROM of a PCI device though.

**Comment by supercomputer7** [comment]
> Apart from those two comments, everything still seems to work with these changes. Have not tested actually dumping the ROM of a PCI device though.

I have tested it with a NVIDIA GT710 card being pass-through'ed to QEMU and it works OK :)
As for "everything seems to work" - well, the actual change is rather small... we are just being more consistent now with the PCI subsystem API, which is great because that means things are more stable, as I mentioned in the commit message.

**Pull request merged**